### PR TITLE
improve k8s-topgun resilience

### PR DESCRIPTION
**Pull request by pivotal-jamie-klassen**
# Existing Issue

Some improvements towards concourse/ci#182. I wouldn't say it completely fixes the issue at this point.

# Changes proposed in this pull request
There are two improvements in this PR:

1. we have found that `kubectl port-forward` seems to intermittently fail, and it has gotten worse lately. Accordingly we will not rely on it to broker our connections to the web nodes during topgun test cases; instead we actually have GKE deploy a load balancer to handle that traffic.
1. the BTRFS cases where failure was expected were making use of the wrong helper function, which meant that they could only pass in certain flaky circumstances.

# Contributor Checklist
- [X] ~Unit tests~
- [X] Integration tests (if applicable)
- [X] ~Updated documentation (located at https://github.com/concourse/docs)~
- [X] ~Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)~

# Reviewer Checklist
> This section is intended for the core maintainers only, to track review progress. Please do not
> fill out this section.
- [X] ~Code reviewed~
- [ ] Tests reviewed
- [X] ~Documentation reviewed~
- [X] ~Release notes reviewed~
- [X] ~PR acceptance performed~

### CODE REVIEW

File: topgun/k8s/baggageclaim_drivers_test.go [hunk 1]
```diff
@@ -72,7 +72,13 @@ func baggageclaimFails(driver string, selectorFlags ...string) {
 	Context(driver, func() {
 		It("fails", func() {
 			setReleaseNameAndNamespace("bd-" + driver)
-			deployWithDriverAndSelectors(driver, selectorFlags...)
+			helmDeployTestFlags := []string{
+				"--set=concourse.web.kubernetes.enabled=false",
```
Comment (jomsie) [inline topgun/k8s/baggageclaim_drivers_test.go]: wondering about this flag, disabling kubernetes just for testing?

File: topgun/k8s/k8s_suite_test.go [hunk 2]
```diff
@@ -251,6 +253,29 @@ func deletePods(namespace string, flags ...string) []string {
...
 
+func getExternalUrl(namespace, resource string) string {
+	protocol := "http"
+	var url string
+	for {
+		session := Start(nil, "kubectl", "get", "svc", "--namespace="+namespace, resource, "-o", "jsonpath='{.status.loadBalancer.ingress[0].ip}'")
+		<-session.Exited
+		s := string(session.Out.Contents())
+		if s != "" {
+			url = protocol + "://" + s[1:len(s)-1] + ":8080"
```
Comment (jomsie) [inline topgun/k8s/k8s_suite_test.go]: figured out how to get around the `'`'s!

File: topgun/k8s/baggageclaim_drivers_test.go [hunk 3]
```diff
@@ -72,7 +72,13 @@ func baggageclaimFails(driver string, selectorFlags ...string) {
...
+				"--set=concourse.worker.baggageclaim.driver=" + driver,
+				"--set=worker.replicas=1",
+			}
+
+			deployFailingConcourseChart(releaseName, "", append(helmDeployTestFlags, selectorFlags...)...)
```
[the remark this answers is not in the document]
Comment (pivotal-jamie-klassen) [reply topgun/k8s/baggageclaim_drivers_test.go]: yes!

File: topgun/k8s/baggageclaim_drivers_test.go (hunk 1 above)
Comment (cirocosta) [reply topgun/k8s/baggageclaim_drivers_test.go]: this disables the use of kubernetes secrets as the credential manager:

https://github.com/helm/charts/blob/2589685b5593c12b448d32b89af65aef70eff10d/stable/concourse/templates/web-deployment.yaml#L370-L384

(see https://concourse-ci.org/kubernetes-credential-manager.html)

Comment (jomsie) [reply topgun/k8s/baggageclaim_drivers_test.go]: nice, thanks ciro!

Comment (pivotal-jamie-klassen) [reply topgun/k8s/baggageclaim_drivers_test.go]: we just inlined `deployWithDriverAndSelectors` here. I'm guessing this has something to do with not using kubernetes to manage secrest.

File: topgun/k8s/k8s_suite_test.go [hunk 4]
```diff
@@ -143,6 +144,7 @@ func helmInstallArgs(args ...string) []string {
 		"--set=web.livenessProbe.initialDelaySeconds=3",
 		"--set=web.livenessProbe.periodSeconds=3",
 		"--set=web.livenessProbe.timeoutSeconds=3",
+		"--set=web.service.type=LoadBalancer",
```
[the remark this answers is not in the document]
Comment (pivotal-jamie-klassen) [reply topgun/k8s/k8s_suite_test.go]: it has never passed

[the remark this answers is not in the document]
Comment (jomsie) [reply topgun/k8s/k8s_suite_test.go]: so for load balancing related quotas, looking here : https://cloud.google.com/load-balancing/docs/quotas its split up into those categories... 

quick look at things we have any usage:
forwarding rules: 48/375

[the remark this answers is not in the document]
Comment (pivotal-bin-ju) [reply topgun/k8s/k8s_suite_test.go]: @cirocosta it is longer than a night to accomplish the 47 specs of k8s topgun.

**Pull request closed without merging**